### PR TITLE
cert-manager: fast-forward to upstream 9a8f4371

### PR DESCRIPTION
**Pull request by munnerz**
* Bump version strings for v0.6.0 (jetstack/cert-manager#1247)
* Set CA sync backoff limit & add verifying installation docs (jetstack/cert-manager#1245)
* Added securityContext to cert-manager deployment pods in helm chart (jetstack/cert-manager#1237)
* Move Helm chart to deploy/charts/cert-manager (jetstack/cert-manager#1235)
* chart: allow deployment strategy customization
* webhook: remove hardcoded image (jetstack/cert-manager#1039)
* Remove crd-install annotation from Challenges and Orders
* Add Challenge and Order CRDs
* Add certificates and issuers to aggregated RBAC for view, edit and admin ClusterRoles. (jetstack/cert-manager#872)
* Use fixed image tag for ca-helper and add description

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

/cc @kragniz

### CODE REVIEW

File: stable/cert-manager/templates/deployment.yaml [hunk 1]
```diff
@@ -14,6 +14,10 @@ spec:
     matchLabels:
       app: {{ template "cert-manager.name" . }}
       release: {{ .Release.Name }}
+  {{- with .Values.strategy }}
+  strategy:
+    {{- . | toYaml | nindent 4 }}
```
Comment (swade1987) [inline stable/cert-manager/templates/deployment.yaml:19]: Think this is meant to say `indent`

**Comment by prydonius** [comment]
@unguiculus looks like the e2e test is trying to install a chart inside the directory of this chart (`stable/cert-manager/webhook`) that shouldn't be installed. Any idea why this might be happening?

**Comment by munnerz** [comment]
To clarify, this sub-chart *should* be installed, and is installed just fine as part of the main cert-manager chart. The issue is as @prydonius says, the e2e script also attempts to install the 'webhook' chart *on its own* (seemingly because it treats the sub-chart as an independent chart)

I guess this is down to the way the chart-testing code discovers charts that need testing?

**Comment by davidkarlsen** [comment]
@munnerz have a look at the failing test?

**Comment by munnerz** [comment]
Just to clarify, this is an example of the actual test failure:
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/helm_charts/10830/pull-charts-e2e/19312/

I added a 'test' commit to try something out, hence the current tip of this
branch is failing for a different reason right now.

But the test failure linked here is the real problem!! 😬

File: stable/cert-manager/templates/deployment.yaml (hunk 1 above)
Comment (munnerz) [reply stable/cert-manager/templates/deployment.yaml:19]: I thought this too 😄 but it in fact adds a newline along with the indent

**Pull request merged**